### DISCUSSION TITLE
What doesn't work/I couldn't/didn't know how to merge/fix

### DISCUSSION THREAD

**Discussion by Panchovix** [Panchovix/stable-diffusion-webui-reForge#11] [General]
Hi there guys, hope is all is going good for you.

I decided after some time not having updates on Forge (~5 months), I felt it was missing important or small updates from A1111, that I should update it so it is more usable and more with the times if it's needed.

So I went, commit by commit from 5 months ago, up to today's updates of the dev branch of A1111 (https://github.com/AUTOMATIC1111/stable-diffusion-webui/commits/dev) and updated the code, manually, from the dev2 branch of forge (https://github.com/lllyasviel/stable-diffusion-webui-forge/commits/dev2) to see which could be merged or not, and which conflicts as well.

​
For now, this is the list of features (probably more) that doesn't work/I couldn't/didn't know how to merge/fix:

* ~~Soft Inpainting (I had to edit sd_samplers_cfg_denoiser.py to apply some A1111 changes, so I couldn't directly apply https://github.com/lllyasviel/stable-diffusion-webui-forge/pull/494) EDIT: For now it seems that soft inpainting is enabled by default, and when checking the option it does another (unkown) thing.
EDIT2: Maybe fixed in https://github.com/Panchovix/stable-diffusion-webui-forge/commit/101af34f83819de8591bdc3c7e11c6fb9466fb03~~
Fixed
* SD3 (Since forge has it's own unet implementation, I didn't tinker on implementing it). Any help to implement this would be really appreciated!
* ~~Callback order (https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/5bd27247658f2442bd4f08e5922afff7324a357a), specifically because the forge implementation of modules doesn't have script_callbacks. So it broke the included controlnet extension and ui_settings.py.~~ Implemented
* ~~Didn't tinker much about changes that affect extensions-builtin\Lora, since forge does it mostly on ldm_patched\modules. But some of them could be applied thay I have may skipped.~~ Implemented
* ~~precision-half (forge has this by default)~~
* ~~New "is_sdxl" flag (sdxl works fine, but there are some new things that don't work without this flag)~~ Implemented
* DDIM CFG++ (because the edit on sd_samplers_cfg_denoiser.py). Researching on how to modify sd_samplers_cfg_denoiser.py to make it work. EDIT: In the progress to make it work.
* ~~https://github.com/Panchovix/stable-diffusion-webui-forge/commit/7f8f332ccafb600e9c7fefa39de44c6b90514626 (NGMI, s_churn, etc is not updated atm because these changed that were needed)~~ Fixed
* Newer controlnet updates into the built-in extension.

The list (but not all) I couldn't/didn't know how to merge/fix is here: https://pastebin.com/sMCfqBua

Any help trying or attempting for the implementation of those is really appreciated!

**Comment by zhaol4v** [comment]
thanks for the hard work, also is batch controlnet supposed to be working now? It still only processes the first image in the specified folder like it used to on main Forge. Maybe I'm doing something wrong though

**Comment by Panchovix** [comment]
> thanks for the hard work, also is batch controlnet supposed to be working now? It still only processes the first image in the specified folder like it used to on main Forge. Maybe I'm doing something wrong though

I haven't touched the integrated controlnet extension for forge at the moment, so probably it works as the same from there. I'm thinking on making just normal controlnet work with forge or seeing how to update the extension to controlnet upstream, but haven't decided yet. It will probably take a while since I haven't seen any of the code of the controlnets extension yet. 

EDIT: Checking any of these options is very hard, so any help will be appreciated since either redirect normal controlnet to forge or updating the in built extension is really, really hard. Really sorry for the ping @huchenlei, but is there a way I can contact you to see this?

Thanks for the info though, gonna add it into the post.

**Comment by StarComrade** [comment]
Would it be possible to revert the Lora preview Tree View to be on the side like it was on the old Forge? Having it on top obscures 80% of the screen and is counterintuitive to the actual previewing.

**Comment by Panchovix** [comment]
> Would it be possible to revert the Lora preview Tree View to be on the side like it was on the old Forge? Having it on top obscures 80% of the screen and is counterintuitive to the actual previewing.

Can you open a new issue with some images please? So I can check what's happening and how I could fix it.

**Comment by angelika3377** [comment]
the refiner isn't working for me at all...

**Reply by Panchovix** [reply]
Do you get any error? I just tested and it seems to apply the refiner correctly.

**Reply by angelika3377** [reply]
yes, TypeError: 'NoneType' object is not iterable, checked with different checkpoints...

**Reply by Panchovix** [reply]
Can you give me the complete error of the console?

**Reply by angelika3377** [reply]
Sure! Thank you for looking into this!

Begin to load 1 model
Reuse 1 loaded models
[Memory Management] Current Free GPU Memory (MB) =  3073.7326793670654
[Memory Management] Model Memory (MB) =  0.0
[Memory Management] Minimal Inference Memory (MB) =  1024.0
[Memory Management] Estimated Remaining GPU Memory (MB) =  2049.7326793670654
Moving model(s) has taken 0.33 seconds
 67%|██████▋   | 20/30 [00:22<00:11,  1.13s/it]
Traceback (most recent call last):
  File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\modules_forge\main_thread.py", line 37, in loop
    task.work()
  File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\modules_forge\main_thread.py", line 26, in work
    self.result = self.func(*self.args, **self.kwargs)
  File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\modules\txt2img.py", line 110, in txt2img_function
    processed = processing.process_images(p)
  File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\modules\processing.py", line 815, in process_images
    res = process_images_inner(p)
  File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\modules\processing.py", line 988, in process_images_inner
    samples_ddim = p.sample(conditioning=p.c, unconditional_conditioning=p.uc, seeds=p.seeds, subseeds=p.subseeds, subseed_strength=p.subseed_strength, prompts=p.prompts)
  File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\extensions\sd-webui-bmab\sd_bmab\sd_override\txt2img.py", line 43, in sample
    samples = self.sampler.sample(self, x, conditioning, unconditional_conditioning, image_conditioning=self.txt2img_image_conditioning(x))
  File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\modules\sd_samplers_kdiffusion.py", line 236, in sample
    samples = self.launch_sampling(steps, lambda: self.func(self.model_wrap_cfg, x, extra_args=self.sampler_extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
  File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\modules\sd_samplers_common.py", line 271, in launch_sampling
    return func()
  File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\modules\sd_samplers_kdiffusion.py", line 236, in <lambda>
    samples = self.launch_sampling(steps, lambda: self.func(self.model_wrap_cfg, x, extra_args=self.sampler_extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
  File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\repositories\k-diffusion\k_diffusion\sampling.py", line 594, in sample_dpmpp_2m
    denoised = model(x, sigmas[i] * s_in, **extra_args)
  File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\modules\sd_samplers_cfg_denoiser.py", line 290, in forward
    if sd_samplers_common.apply_refiner(self, x, sigma):
  File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\modules\sd_samplers_common.py", line 161, in apply_refiner
    timestep = torch.argmin(torch.abs(cfg_denoiser.inner_model.sigmas - torch.max(sigma)))
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
*** Error completing request
*** Arguments: ('task(furfps3y7s19jix)', <gradio.routes.Request object at 0x000001A29F5EFC10>, ' <lora:l1l1c0l3:0.8> Analog style cinematic photograph of l1l1c0l3,  lily cole, she is visting London', '', ['New Default-cinematic'], 1, 1, 9, 750, 700, False, 0.7, 2, 'Latent', 0, 0, 0, 'Use same checkpoint', 'Use same sampler', 'Use same scheduler', '', '', [], 0, 30, 'DPM++ 2M', 'Automatic', True, 'photomatix_pxRealistikaRefiner.safetensors [e2f397721b]', 0.8, -1, False, -1, 0, 0, 0, 0, True, False, {'ad_model': 'face_yolov8n.pt', 'ad_model_classes': '', 'ad_tab_enable': True, 'ad_prompt': 'face of   l1l1c0l3 <lora:l1l1c0l3:0.8> ', 'ad_negative_prompt': '', 'ad_confidence': 0.3, 'ad_mask_k_largest': 0, 'ad_mask_min_ratio': 0, 'ad_mask_max_ratio': 1, 'ad_x_offset': 0, 'ad_y_offset': 0, 'ad_dilate_erode': 4, 'ad_mask_merge_invert': 'None', 'ad_mask_blur': 4, 'ad_denoising_strength': 0.4, 'ad_inpaint_only_masked': True, 'ad_inpaint_only_masked_padding': 32, 'ad_use_inpaint_width_height': False, 'ad_inpaint_width': 512, 'ad_inpaint_height': 512, 'ad_use_steps': False, 'ad_steps': 28, 'ad_use_cfg_scale': False, 'ad_cfg_scale': 7, 'ad_use_checkpoint': False, 'ad_checkpoint': 'Use same checkpoint', 'ad_use_vae': False, 'ad_vae': 'Use same VAE', 'ad_use_sampler': False, 'ad_sampler': 'DPM++ 2M', 'ad_scheduler': 'Use same scheduler', 'ad_use_noise_multiplier': False, 'ad_noise_multiplier': 1, 'ad_use_clip_skip': False, 'ad_clip_skip': 1, 'ad_restore_face': False, 'ad_controlnet_model': 'None', 'ad_controlnet_module': 'None', 'ad_controlnet_weight': 1, 'ad_controlnet_guidance_start': 0, 'ad_controlnet_guidance_end': 1, 'is_api': ()}, {'ad_model': 'None', 'ad_model_classes': '', 'ad_tab_enable': True, 'ad_prompt': '', 'ad_negative_prompt': '', 'ad_confidence': 0.3, 'ad_mask_k_largest': 0, 'ad_mask_min_ratio': 0, 'ad_mask_max_ratio': 1, 'ad_x_offset': 0, 'ad_y_offset': 0, 'ad_dilate_erode': 4, 'ad_mask_merge_invert': 'None', 'ad_mask_blur': 4, 'ad_denoising_strength': 0.4, 'ad_inpaint_only_masked': True, 'ad_inpaint_only_masked_padding': 32, 'ad_use_inpaint_width_height': False, 'ad_inpaint_width': 512, 'ad_inpaint_height': 512, 'ad_use_steps': False, 'ad_steps': 28, 'ad_use_cfg_scale': False, 'ad_cfg_scale': 7, 'ad_use_checkpoint': False, 'ad_checkpoint': 'Use same checkpoint', 'ad_use_vae': False, 'ad_vae': 'Use same VAE', 'ad_use_sampler': False, 'ad_sampler': 'DPM++ 2M', 'ad_scheduler': 'Use same scheduler', 'ad_use_noise_multiplier': False, 'ad_noise_multiplier': 1, 'ad_use_clip_skip': False, 'ad_clip_skip': 1, 'ad_restore_face': False, 'ad_controlnet_model': 'None', 'ad_controlnet_module': 'None', 'ad_controlnet_weight': 1, 'ad_controlnet_guidance_start': 0, 'ad_controlnet_guidance_end': 1, 'is_api': ()}, {'ad_model': 'None', 'ad_model_classes': '', 'ad_tab_enable': True, 'ad_prompt': '', 'ad_negative_prompt': '', 'ad_confidence': 0.3, 'ad_mask_k_largest': 0, 'ad_mask_min_ratio': 0, 'ad_mask_max_ratio': 1, 'ad_x_offset': 0, 'ad_y_offset': 0, 'ad_dilate_erode': 4, 'ad_mask_merge_invert': 'None', 'ad_mask_blur': 4, 'ad_denoising_strength': 0.4, 'ad_inpaint_only_masked': True, 'ad_inpaint_only_masked_padding': 32, 'ad_use_inpaint_width_height': False, 'ad_inpaint_width': 512, 'ad_inpaint_height': 512, 'ad_use_steps': False, 'ad_steps': 28, 'ad_use_cfg_scale': False, 'ad_cfg_scale': 7, 'ad_use_checkpoint': False, 'ad_checkpoint': 'Use same checkpoint', 'ad_use_vae': False, 'ad_vae': 'Use same VAE', 'ad_use_sampler': False, 'ad_sampler': 'DPM++ 2M', 'ad_scheduler': 'Use same scheduler', 'ad_use_noise_multiplier': False, 'ad_noise_multiplier': 1, 'ad_use_clip_skip': False, 'ad_clip_skip': 1, 'ad_restore_face': False, 'ad_controlnet_model': 'None', 'ad_controlnet_module': 'None', 'ad_controlnet_weight': 1, 'ad_controlnet_guidance_start': 0, 'ad_controlnet_guidance_end': 1, 'is_api': ()}, {'ad_model': 'None', 'ad_model_classes': '', 'ad_tab_enable': True, 'ad_prompt': '', 'ad_negative_prompt': '', 'ad_confidence': 0.3, 'ad_mask_k_largest': 0, 'ad_mask_min_ratio': 0, 'ad_mask_max_ratio': 1, 'ad_x_offset': 0, 'ad_y_offset': 0, 'ad_dilate_erode': 4, 'ad_mask_merge_invert': 'None', 'ad_mask_blur': 4, 'ad_denoising_strength': 0.4, 'ad_inpaint_only_masked': True, 'ad_inpaint_only_masked_padding': 32, 'ad_use_inpaint_width_height': False, 'ad_inpaint_width': 512, 'ad_inpaint_height': 512, 'ad_use_steps': False, 'ad_steps': 28, 'ad_use_cfg_scale': False, 'ad_cfg_scale': 7, 'ad_use_checkpoint': False, 'ad_checkpoint': 'Use same checkpoint', 'ad_use_vae': False, 'ad_vae': 'Use same VAE', 'ad_use_sampler': False, 'ad_sampler': 'DPM++ 2M', 'ad_scheduler': 'Use same scheduler', 'ad_use_noise_multiplier': False, 'ad_noise_multiplier': 1, 'ad_use_clip_skip': False, 'ad_clip_skip': 1, 'ad_restore_face': False, 'ad_controlnet_model': 'None', 'ad_controlnet_module': 'None', 'ad_controlnet_weight': 1, 'ad_controlnet_guidance_start': 0, 'ad_controlnet_guidance_end': 1, 'is_api': ()}, False, 'Use same checkpoint', 'Use same vae', 1, 0, 'None', 'None', 'None', False, 0.15, 3, 0.4, 4, 'bicubic', 0.5, 2, True, False, True, False, False, False, 'Use same checkpoint', 'Use same vae', 'txt2img-1pass', 'None', '', '', 'Use same sampler', 'Use same scheduler', 'BMAB fast', 20, 7, 0.75, 0.5, 0.1, 0.9, False, False, 'Use same checkpoint', 'Use same vae', 'Select Model', 'None', '', '', 'Use same sampler', 'Use same scheduler', 20, 7, 0.75, 4, 0.35, False, 50, 200, 0.5, False, True, 'stretching', 'bottom', 'None', 0.85, 0.75, False, 'Use same checkpoint', 'Use same vae', True, '', '', 'Use same sampler', 'Use same scheduler', 'BMAB fast', 20, 7, 0.75, 1, 0, 0, 1, 1, 1, 1, 0, 0, 0, None, False, 1, False, '', False, False, False, True, True, 'Use same checkpoint', 'Use same vae', 'Use same sampler', 'Use same scheduler', 4, 3, 0.1, 1, 1, 0, 0.4, 7, 20, False, False, False, 'Score', 1, '', '', '', '', '', '', '', '', '', '', False, 512, 512, 7, 20, 4, 'Use same checkpoint', 'Use same vae', 'Use same sampler', 'Use same scheduler', 'Only masked', 32, 'BMAB Face(Normal)', 0.4, 4, 0.35, False, 0.26, False, True, False, 'subframe', '', '', 0.4, 7, True, 4, 0.3, 0.1, 'Only masked', 32, '', False, False, False, 0.4, 0.1, 0.9, 'Both', False, 1, 0, 1, False, 'Random', False, 0.6, 0, 0.3, 'Random', 'normal', False, True, 'normal', '', 'None', 0.5, False, False, 'Inpaint', 0.85, 0.6, 30, False, True, 'None', 1.5, 'None', False, 'AGENCYB.TTF', 'bottom-left', 'left', '0', '#000000', '#000000', 12, 100, 0, 5, '', '', 'None', ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=[], batch_mask_gallery=[], multi_inputs_gallery=[], generated_image=None, mask_image=None, hr_option=<HiResFixOption.BOTH: 'Both'>, enabled=False, module='None', model='None', weight=1, image=None, resize_mode=<ResizeMode.INNER_FIT: 'Crop and Resize'>, processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0, guidance_end=1, pixel_perfect=False, control_mode=<ControlMode.BALANCED: 'Balanced'>, save_detected_map=True), ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=[], batch_mask_gallery=[], multi_inputs_gallery=[], generated_image=None, mask_image=None, hr_option=<HiResFixOption.BOTH: 'Both'>, enabled=False, module='None', model='None', weight=1, image=None, resize_mode=<ResizeMode.INNER_FIT: 'Crop and Resize'>, processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0, guidance_end=1, pixel_perfect=False, control_mode=<ControlMode.BALANCED: 'Balanced'>, save_detected_map=True), ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=[], batch_mask_gallery=[], multi_inputs_gallery=[], generated_image=None, mask_image=None, hr_option=<HiResFixOption.BOTH: 'Both'>, enabled=False, module='None', model='None', weight=1, image=None, resize_mode=<ResizeMode.INNER_FIT: 'Crop and Resize'>, processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0, guidance_end=1, pixel_perfect=False, control_mode=<ControlMode.BALANCED: 'Balanced'>, save_detected_map=True), False, 7, 1, 'Constant', 0, 'Constant', 0, 1, 'enable', 'MEAN', 'AD', 1, False, 1.01, 1.02, 0.99, 0.95, False, 0.5, 2, False, 256, 2, 0, False, False, 3, 2, 0, 0.35, True, 'bicubic', 'bicubic', False, 0, 'anisotropic', 0, 'reinhard', 100, 0, 'subtract', 0, 0, 'gaussian', 'add', 0, 100, 127, 0, 'hard_clamp', 5, 0, 'None', 'None', False, 'MultiDiffusion', 768, 768, 64, 4, False, False, False, False, False, 'positive', 'comma', 0, False, False, 'start', '', 1, '', [], 0, '', [], 0, '', [], True, False, False, False, False, False, False, 0, False) {}
    Traceback (most recent call last):
      File "C:\Users\user\AppData\Roaming\StabilityMatrix\Packages\Stable Diffusion WebUI Forge\modules\call_queue.py", line 58, in f
        res = list(func(*args, **kwargs))
    TypeError: 'NoneType' object is not iterable

I have copied your whole rep into the Stability Matrix folder :)

**Reply by Panchovix** [reply]
By any change, are you using SD 2.x or 1.5 models? Besides that, can you try the same workflow you're doing but with reForge directly? (I don't know what Stability Matrix is sadly)

**Reply by angelika3377** [reply]
it's just a library that shares folders with other applications, it provides the whole structure, I don't think I can change it... there is already a structure in my data, different folders etc. it doesn't do anything else. yes, this is SD 1.5

**Reply by Panchovix** [reply]
Okay I should have fixed it on last commit

https://github.com/Panchovix/stable-diffusion-webui-reForge/commit/d8d381cf9974ccf9a9c98c461db38b0e780182b1

I don't use refiner often, but at least with SD1.5 base and SD1.5 refiner, works fine. SDXL base and SDXL refiner, same as well.

When mixing architectures it seems to have some issues because the VAE, does A1111 handle that?

**Reply by angelika3377** [reply]
thank you it works :) I'm pretty sure A1111 handled that, but I've been using Forge for a long time

**Comment by amitsandhel** [comment]
When I ran this last time it was slower then forge on my nvidia gtx1050Ti 4GB VRAm graphic card 
is this branch supposed to be as fast as forge on weaker low VRAM hardware since you took the branch from forge right ?

Apologies for posting here I wasn't sure where the correct place to post would be

**Reply by Panchovix** [reply]
Hi there, when it was that last time? If it was today, it depends of the branch you're using.

If main, that is mostly A1111 changes and small model management changes for multiple checkpoints.
If dev_upstream, that's a lot of backend changes that could change that.

And yes it should be the as fast as original Forge, but I haven't tested on weaker VRAM hardware at the moment.

I did create some experimental branches (experimental_main and experimental_dev_upstream) that have the original forge model management. You can try those if you want to see if you get the expected performance. But those branches may have some issues with other (new mostly) additions of reforge.

**Reply by amitsandhel** [reply]
Hello

It was maybe 2 days ago when I first tried to run your system. Today I tried it again on the dev-upstream branch and I am getting an iteration time of 9sec/iter which is similar to what i get with forge. 

The only problem is that highres fix is broken. When I do high res fix even at a resolution size of 1 it goes into OOM (out of memory) at the end. 

If you want I am willing to test your changes on my low end graphic card to see how it performs there ?

**Reply by Panchovix** [reply]
Yeah sure, if it fails at the end it's probably something related to the VAE. Can you print the error? Or still outputs but slowly?

You can also try main branch, since dev_upstream can be a bit experimental with some comfy upstream changes

**Comment by amitsandhel** [comment]
Hello a question 

Does controlnet for sdxl loras and checkpoints work in this version of forge by any chance 
and which sdxl controlnet repo is the best or recommended for us to use 

thanks

**Reply by Panchovix** [reply]
Hi there, honestly I'm not sure since I'm new to controlnet and pushed fixes mostly to it, and updates on the dev_upstream branch. You can try the ones you want to use and let me know if you have any issues.

**Comment by CosmicElement** [comment]
Hiya, not sure if this is worth it's own issue, but I ran into some issues when trying to switch my Auto1111 install to reForge. I followed the commands in the readme and when I tried git pull, at first it gave me this.
```
Committer identity unknown

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'whatever')
```
After I set my name and email I tried again and got a bunch of merge conflicts. (This isn't the whole thing, just the last few lines)
```
Auto-merging pyproject.toml
Auto-merging requirements.txt
CONFLICT (content): Merge conflict in requirements.txt
Auto-merging requirements_versions.txt
CONFLICT (content): Merge conflict in requirements_versions.txt
Auto-merging scripts/postprocessing_upscale.py
CONFLICT (content): Merge conflict in scripts/postprocessing_upscale.py
Auto-merging scripts/xyz_grid.py
CONFLICT (content): Merge conflict in scripts/xyz_grid.py
Auto-merging style.css
CONFLICT (content): Merge conflict in style.css
Auto-merging webui.sh
CONFLICT (content): Merge conflict in webui.sh
Automatic merge failed; fix conflicts and then commit the result.
```
Any help would be appreciated.

**Reply by Panchovix** [reply]
Did this happen when using git reset --hard, or before?

**Reply by CosmicElement** [reply]
After git reset --hard, did the commands in the order they were on the page.

**Reply by Panchovix** [reply]
Can you try with the same steps, but instead of that, do "git stash" instead of "git reset --hard"?

`git reset --hard HEAD@{1}` should let you go to a commit before the git reset --hard

**Reply by CosmicElement** [reply]
When I try git stash it just says "No local changes to save"

**Reply by Panchovix** [reply]
And git pull doesn't work either? Anyways, it is really weird that for some reason it asked your email, that shouldn't have happened :(.

It is doable to the process again from the start? If not, I would suggest to do a clean "git clone" on a new folder and move A1111 files like models, configs, embeddings etc into it.

**Reply by CosmicElement** [reply]
git pull gives something different this time. I may have to start a new install anyways, getting a little old. Any idea why it might've asked for my email?
```
error: Pulling is not possible because you have unmerged files.
hint: Fix them up in the work tree, and then use 'git add/rm <file>'
hint: as appropriate to mark resolution and make a commit.
fatal: Exiting because of an unresolved conflict.
```

**Reply by Panchovix** [reply]
`git merge --abort` should maybe work and then git stash, then git pull.

If not, then cloning in a new instance would work.

Also, I'm not sure why it asked for your email,  that shouldn't happened. I formatted my PC and when cloning I didn't got asked to use my github email D:

**Reply by CosmicElement** [reply]
Just wanted to follow up, I think the reason my update didnt work was because my A1111 install was from the zip package rather than a git clone, so I think the error was on my part.

**Comment by amitsandhel** [comment]
@Panchovix  
your latest change or update on dev-upstream broke it that it won't even run that's the error I'm getting 

![image](https://github.com/user-attachments/assets/66303a39-0862-4e46-814e-8e5458f4470b)


this is the update I did today 
![image](https://github.com/user-attachments/assets/d31bf337-c429-4c72-bfbd-d1c2e62a09f0)

**Reply by Panchovix** [reply]
Thanks for the report, I pushed some commits that should fix it.

**Comment by amitsandhel** [comment]
@Panchovix  on the dev_upstream branches your new updates broke I get an error about comfy it doesn't crash just kind of gives a warning though in my console this one 

![image](https://github.com/user-attachments/assets/ae8bb9dc-b6ad-4352-ae84-9d3ea15ead68)

these were the git fetch and pull changes you did 
![image](https://github.com/user-attachments/assets/65974466-257d-4244-8581-3cbd838740ed)


do note that I am using the simple RAUNET-MSW-MSA for forge setting

**Reply by Panchovix** [reply]
Now it should be fixed, my bad!

**Comment by amitsandhel** [comment]
@Panchovix  just did a git fetch and pull on your new stuff seems there is another error
as an aside do you want me to help you unittest or test this stuff out before you deploy it to make sure there are no errors 
i'm looking for a project anyways so i can be happy to help you if you want ?
![image](https://github.com/user-attachments/assets/fa2af9bd-8154-41e6-8204-01199f97cbf8)

**Reply by Panchovix** [reply]
Okay that was dumb haha I forgot to add the req for Automatic CFG, it should be fixed now.

About testing, it could be, though since it is a dev branch it is expected to have some issues. I maybe could create a branch named "dev_experimental" where I push the changes first, and if it works, I push to "dev_upstream"

**Comment by amitsandhel** [comment]
@Panchovix  
so the dev/upstream unit after todays update is broken again I can't even run the server. 
There's a screenshot below of what happened
![image](https://github.com/user-attachments/assets/398d120e-2606-4d5f-9ba6-bba9a5642d9a)

**Reply by Panchovix** [reply]
I think it's still missing a bit of the error, can you send the complete log?

**Reply by Panchovix** [reply]
Pushed/reverted/reapplies some changes (that shouldn't degrade performance if low vram), can you check if it works fine for you?

**Reply by amitsandhel** [reply]
@Panchovix  Yes it seems to be working now at least the browser is loading now I'm running a test to see if I can generate images if any issues come up I'll post them here for you

Thanks

**Comment by amitsandhel** [comment]
@Panchovix  on the main/newforge thread 
if i use controlenet with say depth and a small depth midas model I can generate an image with a 832x1216 image. But if I use high-res fix i find that it goes into OOM and gives me the error below ?

![image](https://github.com/user-attachments/assets/5391d39b-6e12-4daf-b242-900f3160021e)

![image](https://github.com/user-attachments/assets/bdab87ef-7df5-47fe-a326-ff80823b82d6)

**Reply by Panchovix** [reply]
For this branch, since it is exactly as the OG forge with added samplers, you can report that issue on the OG forge issues (https://github.com/lllyasviel/stable-diffusion-webui-forge/issues) if you aren't using any of the new samplers.

**Comment by amitsandhel** [comment]
@Panchovix  
i just did a git fetch and pull on your dev_upstream branch and I got the following error it won't even run the program:

**stable-diffusion-webui-reForge/ldm_patched/modules/model_management.py", line 179, in get_optimal_vae_dtype
    elif torch.cuda.is_fp16_supported():
AttributeError: module 'torch.cuda' has no attribute 'is_fp16_supported'. Did you mean: 'is_bf16_supported'?**

**Reply by Panchovix** [reply]
Which torch version do you have? Also can you post the complete error?

**Reply by amitsandhel** [reply]
@Panchovix 
![image](https://github.com/user-attachments/assets/06c1b79e-8bf7-46d9-bfd5-e8205b9990f8)
 
how can i check which torch version i have hopefully the image above helps

**Reply by Panchovix** [reply]
Okay I can see it now, pushed a commit that should fix this https://github.com/Panchovix/stable-diffusion-webui-reForge/commit/ece1253051bedfac28a5f941a22720e59a0427cb

Can you try and see if it works fine?

Not needed now, but to check your python version, you can run `pip list` inside your venv, and then see the torch version there.

**Reply by amitsandhel** [reply]
@Panchovix  
okay my torch version is torch 2.3.1
and yes now its working thanks

**Comment by amitsandhel** [comment]
@Panchovix  
it seems the UI is working but again on the dev/upstream branch it isn't picking up cuda at all ?
![image](https://github.com/user-attachments/assets/b2f0f909-e768-4256-8d87-af43fd65bbac)

**Reply by Panchovix** [reply]
That seems to be an specific bitsandbytes thing? In theory you should be able to do inference normally.

Did you install a new extension?

I haven't made a change since the fix from yesterday up to today.

**Comment by amitsandhel** [comment]
@Panchovix  did you update the dev_upstream branch or even work on this branch or are we jumping back to forge again ?
I had some laptop updates and had to update my driver but i get this error today when I run it ?
![image](https://github.com/user-attachments/assets/24b29456-05a4-4f95-88b8-f498dbdaf208)

**Reply by Panchovix** [reply]
Hi there, I'm in holidays so I haven't check the repo most of the time. I haven't updated anything yet since all was stable.

**Reply by amitsandhel** [reply]
@Panchovix  
i fixed the issue it seemed I updated my nvidia-driver to 560 but that caused issue so after some fixes it works though who knows I'm still testing it but at least it is working now 
thanks

**Comment by ArnorWing** [comment]
@Panchovix 
This fine gentleman came up with a fix for controlnet for Forge main. 
Any chance of having a look and incorporating into reForge?
https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/287#issuecomment-2416988980
